summary/src/main/scala/JobUtils.scala

```diff
@@ -235,16 +235,16 @@ trait JobUtils {
     * Set of methods for easy processing of parameters
     */
   def toLayers(rasterLayerIds: Seq[LayerId], polygon: Seq[MultiPolygon], sc: SparkContext) = {
-    val extent = GeometryCollection(polygon).envelope
+    val multiPolygon = polygon.unionGeometries.asMultiPolygon.get
     val rasterLayers = rasterLayerIds.map({ rasterLayerId =>
-      queryAndCropLayer(catalog(sc), rasterLayerId, extent)
+      queryAndCropLayer(catalog(sc), rasterLayerId, multiPolygon)
     })
 
     rasterLayers
   }
 
   def toLayers(rasterLayerIds: Seq[LayerId], targetLayerId: LayerId, polygon: Seq[MultiPolygon], sc: SparkContext) = {
-    val extent = GeometryCollection(polygon).envelope
+    val extent = polygon.unionGeometries.asMultiPolygon.get
```

Comment:
Should the `val` name here also be changed from `extent` -> `multiPolygon` to match the other changes?